src/platforms/iOS.ts

```diff
@@ -0,0 +1,63 @@
+import { join } from 'node:path'
+import { readFileSync } from 'node:fs'
+import type { GenerateNotesContext } from 'semantic-release'
+
+export interface IOSPlatformConfiguration {
+  podSpecJsonPath: string | undefined
+  dependencyName: string | undefined
+  displayName: string | undefined
+}
+
+type PodspecJson = {
+  dependencies: {
+    [key: string]: [string]
+  }
+}
+
+export type IOSResolveContext = Pick<GenerateNotesContext, 'cwd'>
+
+export const resolve = async (
+  { cwd }: IOSResolveContext,
+  { podSpecJsonPath, dependencyName }: IOSPlatformConfiguration
+) => {
+  if (!cwd) {
+    throw new Error(`Current working directory is required to detect iOS dependency version range.`)
+  }
+
+  if (!podSpecJsonPath) {
```

Comment:
Similar to android, let's make these required properties required in `IOSPlatformConfiguration` as well.